pkg/cmd/server/crypto/crypto.go

```diff
@@ -542,20 +558,37 @@ func CertsFromPEM(pemCerts []byte) ([]*x509.Certificate, error) {
 }
 
 // Can be used as a certificate in http.Transport TLSClientConfig
-func newClientCertificateTemplate(subject pkix.Name) (*x509.Certificate, error) {
+func newClientCertificateTemplate(subject pkix.Name, expireDays int, currentTime func() time.Time) *x509.Certificate {
+	var lifetimeInDays = DefaultCertificateLifetimeInDays
+	if expireDays > 0 {
+		lifetimeInDays = expireDays
+	}
+
+	if lifetimeInDays > DefaultCertificateLifetimeInDays {
+		warnAboutCertificateLifeTime(subject.CommonName, DefaultCertificateLifetimeInDays)
+	}
+
+	lifetime := time.Duration(lifetimeInDays) * 24 * time.Hour
+
 	return &x509.Certificate{
 		Subject: subject,
 
 		SignatureAlgorithm: x509.SHA256WithRSA,
 
-		NotBefore:    time.Now().Add(-1 * time.Second),
-		NotAfter:     time.Now().Add(lifetime),
+		NotBefore:    currentTime().Add(-1 * time.Second),
+		NotAfter:     currentTime().Add(lifetime),
 		SerialNumber: big.NewInt(1),
 
 		KeyUsage:              x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature,
 		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth},
 		BasicConstraintsValid: true,
-	}, nil
+	}
+}
+
+func warnAboutCertificateLifeTime(name string, defaultLifetimeInDays int) {
```

Comment:
Yes, exactly as you said, it will be printed ~10 times. (And I put this example to the PR description.)

To avoid such behavior we should add one more parameter (for example, `suppressWarning`) to the methods. Shall I do it?